src/realm/table.hpp

```diff
@@ -503,6 +506,22 @@ class Table {
     void set_null(size_t column_ndx, size_t row_ndx, bool is_default = false);
     void set_null_unique(size_t col_ndx, size_t row_ndx);
 
+    template <class T>
+    void set(size_t c, size_t r, T value, bool is_default = false);
+
+    template <class T>
+    void set_list(size_t c, size_t r, const std::vector<T>& list);
+
+    template <typename T>
+    class List;
+
+    /// The object returned here is only valid during the current transaction
+    /// It cannot be updated or handed over.
+    template <class T>
+    std::unique_ptr<List<T>> get_list(size_t column_ndx, size_t row_ndx);
```

Comment:
First of all to make an interface similar to the link list interface (Table::get_linklist()). I was first more inclined to just return the object by value, but that would not really support returning a const object if the table is const. I don't suppose you would prefer returning a raw pointer.